### DISCUSSION TITLE
Warning message: In readLines(script) : incomplete final line found on '_targets.R'

### DISCUSSION THREAD

**Discussion by gorkang** [ropensci/targets#565] [Help] [answered]
Just updated targets to 0.6.0, and suddenly I am seeing this warning in all my projects:

```
Warning message:
In readLines(script) : incomplete final line found on '_targets.R'
```
The structure of my _targets.R file is something like:

```
targets <- list(
  tar_target(...)
)
targets # This is the last line
```

Any way to avoid this warning? Thanks!

**Comment by wlandau** [comment, marked as answer]
The warning comes directly from `readLines()`. You can avoid it by making sure your `_targets.R` file ends with a line break. Related: https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

**Reply by gorkang** [reply]
So sorry. I interpreted the warning the other way around. I can see having an extra space in that last empty line also causes issues. Thanks!

**Comment by dkibalnikov** [comment]
I just leave last line as empty one and warning is disappeared.
